### PR TITLE
Fix filtering and lifting by a tag with a variable

### PR DESCRIPTION
**Pull request by jnwatson**

### CODE REVIEW

File: synapse/lib/ast.py [hunk 1]
```diff
@@ -1299,50 +1300,84 @@ class TagCond(Cond):
...
-    def getCondEval(self, runt):
+    async def getCondEval(self, runt):
 
-        name = self.kids[0].value()
+        assert len(self.kids) == 1
+        kid = self.kids[0]
+        if isinstance(kid, TagMatch):
+            name = self.kids[0].value()
+        else:
+            if kid.isRuntSafe(runt):
```
Comment (invisig0th) [inline synapse/lib/ast.py]: heh, if / else block that do the same thing :)

File: synapse/lib/ast.py [hunk 2]
```diff
@@ -1299,50 +1300,84 @@ class TagCond(Cond):
...
 
             return cond
 
-        # Default exact match
+        # kid is a non-runtsafe VarValue: dynamically evaluate value of variable for each node
         async def cond(node, path):
+            name = await kid.compute(path)
+
+            if name == '*':
+                return bool(node.tags)
```
Comment (invisig0th) [inline synapse/lib/ast.py]: <3 avoidance of conditional logic

File: synapse/lib/ast.py [hunk 3]
```diff
@@ -1299,50 +1300,82 @@ class TagCond(Cond):
...
+            # if kid.isRuntSafe(runt):
+            #     name = await kid.runtval(runt)
+            name = None
+
+        if name is not None:
+
+            # Allow for a user to ask for #* to signify "any tags on this node"
+            if name == '*':
+                async def cond(node, path):
+                    # Check if the tags dictionary has any members
```
Comment (vEpiphyte) [inline synapse/lib/ast.py]: this could also be `return bool(node.tags)`

Comment (jnwatson) [reply synapse/lib/ast.py]: done

**Pull request merged**